src/main/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/dwpuploadresponse/DwpUploadResponseMidEventValidationHandler.java

```diff
@@ -0,0 +1,80 @@
+package uk.gov.hmcts.reform.sscs.ccd.presubmit.dwpuploadresponse;
+
+import static uk.gov.hmcts.reform.sscs.ccd.presubmit.dwpuploadresponse.ElementsDisputed.*;
+
+import java.util.Objects;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Component;
+import uk.gov.hmcts.reform.sscs.ccd.callback.Callback;
+import uk.gov.hmcts.reform.sscs.ccd.callback.CallbackType;
+import uk.gov.hmcts.reform.sscs.ccd.callback.PreSubmitCallbackResponse;
+import uk.gov.hmcts.reform.sscs.ccd.domain.EventType;
+import uk.gov.hmcts.reform.sscs.ccd.domain.SscsCaseData;
+import uk.gov.hmcts.reform.sscs.ccd.presubmit.PreSubmitCallbackHandler;
+
+@Component
+@Slf4j
+public class DwpUploadResponseMidEventValidationHandler implements PreSubmitCallbackHandler<SscsCaseData> {
+
+    PreSubmitCallbackResponse<SscsCaseData> preSubmitCallbackResponse;
+
+    @Override
+    public boolean canHandle(CallbackType callbackType, Callback<SscsCaseData> callback) {
+        return callbackType == CallbackType.MID_EVENT
+            && callback.getEvent() == EventType.DWP_UPLOAD_RESPONSE
+            && Objects.nonNull(callback.getCaseDetails())
+            && Objects.nonNull(callback.getCaseDetails().getCaseData());
+    }
+
+    @Override
+    public PreSubmitCallbackResponse<SscsCaseData> handle(CallbackType callbackType, Callback<SscsCaseData> callback, String userAuthorisation) {
+        if (!canHandle(callbackType, callback)) {
+            throw new IllegalStateException("Cannot handle callback");
+        }
+
+        SscsCaseData sscsCaseData = callback.getCaseDetails().getCaseData();
+
+        preSubmitCallbackResponse = new PreSubmitCallbackResponse<>(sscsCaseData);
+
+        checkForDuplicateIssueCodes(sscsCaseData);
+
+        return preSubmitCallbackResponse;
+    }
+
+    private void checkForDuplicateIssueCodes(SscsCaseData sscsCaseData) {
+        String duplicateMessage = " element contains duplicate issue codes";
+
+        if (sscsCaseData.getElementsDisputedGeneral() != null
+                && sscsCaseData.getElementsDisputedGeneral().size() != sscsCaseData.getElementsDisputedGeneral().stream().distinct().count()) {
```

Comment:
These blocks could be refactored to use a common method - eg.
```
private void validateElementDisputedList(List<ElementDisputed> list, ElementsDisputed elementsDisputedType) {
        String duplicateMessage = " element contains duplicate issue codes";
        if (list != null
            && list.size() != list.stream().distinct().count()) {
            preSubmitCallbackResponse.addError(elementsDisputedType.getValue() + duplicateMessage);
        }
    }
```
Think that would be clearer and more readable.

Once it's refactored like this, could potentially create an @DistinctIssueCodeElementDisputedList attonation validator in common that could be added to each field ( if it was something that could be reused) .  If it's not something that'll be reused then maybe not worth creating the validator - up to you

Reply:
Looks better, thanks